memcache/memcache.go

```diff
@@ -267,7 +383,14 @@ func (c *Client) dial(addr net.Addr) (net.Conn, error) {
 	return nil, err
 }
 
-func (c *Client) getConn(addr net.Addr) (*conn, error) {
+func (c *Client) getConn(ctx context.Context, addr net.Addr) (*conn, error) {
+	// Check if the context is expired.
+	select {
+	default:
+	case <-ctx.Done():
```

Comment:
This looks a bit useless. At this point there is no network call yet. Dial happens below in c.dial. You want to extend dial with custom dialer, see https://pkg.go.dev/net#Dialer.DialContext